src/client/QXmppRosterManager.cpp

```diff
@@ -234,6 +234,7 @@ void QXmppRosterManager::_q_presenceReceived(const QXmppPresence &presence)
             subscribe(bareJid);
         } else {
             emit subscriptionReceived(bareJid);
+            emit subscriptionRequestReceived(bareJid, presence.statusText());
```

Comment:
To make this future proof, let's include the QXmppPresence and a unit test would be nice of course. :)